core/src/main/scala/zio/jdbc/ZConnection.scala

```diff
@@ -25,53 +25,76 @@ import java.sql.{ Connection, PreparedStatement, Statement }
  * `Connection` through the `access` method. Any such access will be attempted on the
  * blocking thread pool.
  */
-final class ZConnection(private[jdbc] val connection: Connection) extends AnyVal {
-  def access[A](f: Connection => A): ZIO[Any, Throwable, A] = ZIO.attemptBlocking(f(connection))
+final class ZConnection(private[jdbc] val stateful: ZStatefulConnection) extends AnyVal {
 
-  def accessZIO[A](f: Connection => ZIO[Scope, Throwable, A]): ZIO[Scope, Throwable, A] = ZIO.blocking(f(connection))
+  private[jdbc] def access[A](f: Connection => A): ZIO[Any, Throwable, A] =
+    ZIO.attemptBlocking(f(stateful.underlying))
 
-  def close: Task[Any]    = access(_.close())
-  def rollback: Task[Any] = access(_.rollback())
+  private[jdbc] def accessZIO[A](f: Connection => ZIO[Scope, Throwable, A]): ZIO[Scope, Throwable, A] =
+    ZIO.blocking(f(stateful.underlying))
+
+  def close: Task[Any] =
+    access(_.close())
+
+  def commit: Task[Unit] =
+    access(_.commit())
 
   private[jdbc] def executeSqlWith[A](
     sql: SqlFragment
-  )(f: PreparedStatement => ZIO[Scope, Throwable, A]): ZIO[Scope, Throwable, A] =
-    accessZIO { connection =>
-      for {
-        statement <- ZIO.acquireRelease(ZIO.attempt {
-                       val sb = new StringBuilder()
-                       sql.foreachSegment(syntax => sb.append(syntax.value))(_ => sb.append("?"))
-                       connection.prepareStatement(sb.toString, Statement.RETURN_GENERATED_KEYS)
-                     })(statement => ZIO.attemptBlocking(statement.close()).ignoreLogged)
-        _         <- ZIO.attempt {
-                       var paramIndex = 1
-                       sql.foreachSegment(_ => ()) { param =>
-                         param.setter.setValue(statement, paramIndex, param.value)
-                         paramIndex += 1
-                       }
-                     }
-        result    <- f(statement)
-      } yield result
-    }.tapErrorCause { cause => // TODO: Question: do we want logging here, switch to debug for now
+  )(f: PreparedStatement => ZIO[Scope, Throwable, A]): ZIO[Scope with Option[TransactionIsolationLevel], Throwable, A] =
+    (for {
+      isolationLevel <- ZIO.service[Option[TransactionIsolationLevel]]
+      statement      <- ZIO.acquireRelease(
+                          for {
+                            _  <- isolationLevel match {
+                                    case Some(level) =>
+                                      stateful.setTransactionIsolation(level) *> stateful.setAutoCommit(false)
```

Comment:
These methods are optimized in a way that we don't call methods of JDBC connection when the given values as in the current state of `ZStatefulConnection`. It means that each query in a transaction won't set value again and again